src/pages/resultsView/mutation/MutationMapperStore.ts

```diff
@@ -142,10 +143,33 @@ export class MutationMapperStore {
         }
     }, undefined);
 
+    readonly allTranscripts = remoteData<EnsemblTranscript[] | undefined>({
+        invoke: async()=>{
+            if (this.gene) {
+                return fetchEnsemblTranscriptsByEnsemblFilter({"hugoSymbols":[this.gene.hugoGeneSymbol]});
+            } else {
+                return undefined;
+            }
+        },
+        onError: (err: Error) => {
+            console.log("Getting all transcripts failed");
+        }
+    }, undefined);
+
     readonly canonicalTranscript = remoteData<EnsemblTranscript | undefined>({
+        await: () => [
+            this.allTranscripts
+        ],
         invoke: async()=>{
             if (this.gene) {
-                return fetchCanonicalTranscript(this.gene.hugoGeneSymbol, this.isoformOverrideSource);
+                return fetchCanonicalTranscript(this.gene.hugoGeneSymbol, this.isoformOverrideSource).catch(async() => {
+                    console.log("No canonical transcript for: " + this.gene.hugoGeneSymbol);
+                    const transcript = _.maxBy(this.allTranscripts.result, (t:EnsemblTranscript) => t.proteinLength);
+                    if (transcript) {
+                        console.log("Using transcript with longest length instead: " + transcript);
```

Comment:
I have always had mixed feelings about `console.log` :)

Reply:
same, if we have a better way of logging warnings would be good